pymc3/distributions/distribution.py

```diff
@@ -355,7 +357,9 @@ def _draw_value(param, point=None, givens=None):
         if point and hasattr(param, 'model') and param.name in point:
             return point[param.name]
         elif hasattr(param, 'random') and param.random is not None:
-            return param.random(point=point, size=None)
+            return param.random(point=point, size=None).mean(axis=0)
```

Comment:
this `mean` here is causing the strangeness.  In your test, it is just drawing 10 values for `a`, taking the mean (which is likely reasonably close to 0.5), and then using that value for all 10k draws of `b`.

Reply:
Let me try removing the mean :)